### PR TITLE
Update types for dynamodb-lock-client

### PR DESCRIPTION
**Pull request by WilliamHo-at**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://github.com/tristanls/dynamodb-lock-client/blob/f8cb1bc4d40eab6962875fe60a7e02c09d09b1ce/README.md#dynamodblockclient>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## dynamodb-lock-client ([<kbd>unpkg</kbd>](https://unpkg.com/browse/dynamodb-lock-client@latest/))
was missing the following properties:
1. Lock
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against d9d937c783d79923b9a35864d7ea2366d2d17ef2
</p>

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## dynamodb-lock-client ([<kbd>unpkg</kbd>](https://unpkg.com/browse/dynamodb-lock-client@latest/))
was missing the following properties:
1. Lock
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 852e58ae8ba0e1762bf51cb1828b4eede5b0734e
</p>

**Comment by peterblazejewicz** [comment]
@WilliamHo-at can you please, in the absence of owners review, link the source/documentation for the changes covered in PR? thanks!

**Comment by WilliamHo-at** [comment]
> @WilliamHo-at can you please, in the absence of owners review, link the source/documentation for the changes covered in PR? thanks!

I based my changes on the README.md for this package: https://github.com/tristanls/dynamodb-lock-client/blob/master/README.md#clientacquirelockid-callback. The commit that added these changes is https://github.com/tristanls/dynamodb-lock-client/commit/4fd86ce62998679e92ecf6d46a575cd8bf92f464

File: types/dynamodb-lock-client/index.d.ts [hunk 1]
```diff
@@ -34,15 +35,15 @@ export interface FailOpenConfig extends GenericConfig {
     trustLocalTime?: boolean | undefined;
 }
 
-export class LockClient<PartitionTableKeyType extends string | number> {
+export class LockClient<PartitionTableKeyType extends string | number | Buffer | object> {
```
Comment (peterblazejewicz) [inline types/dynamodb-lock-client/index.d.ts:38]: so those objects are more like:
```suggestion
export class LockClient<PartitionTableKeyType extends string | number | Buffer | Record<string | Buffer | number>> {
```

>  If the type of id is Object, it is expected to have the following format:
{
  [config.partitionKey]: String|Buffer|Number,
  [config.sortKey]: String|Buffer|Number
}

Comment (peterblazejewicz) [reply types/dynamodb-lock-client/index.d.ts:38]: I mean: nearly everything is an object, here the shape for the object should be very specific

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@WilliamHo-at thanks!

**Comment by WilliamHo-at** [comment]
Ready to merge

**Pull request merged**